src/pointer.rs

```diff
@@ -52,13 +52,20 @@ impl Pointer {
     ///
     /// This is a zero-copy constructor
     fn new<S: AsRef<str> + ?Sized>(s: &S) -> &Self {
-        unsafe { &*(core::ptr::from_ref::<str>(s.as_ref()) as *const Self) }
+        // unsafe { &*(core::ptr::from_ref::<str>(s.as_ref()) as *const Self) }
+        #[allow(clippy::ref_as_ptr)]
```

Comment:
Did this break something? I saw that the lint was a bit wonky (alternated between being too strict and too lax) from PR to PR.

Reply:
I rolled it back to make sure it wasn't the problem. I also didn't want the change that clippy made to be a cause of confusion since you had initially wrote it the other way.

I'll revert it back to the way it was before.

Reply:
Sorry for adding so much noise to your diagnostics.

Reply:
Makes sense, no worries. 👍

Reply:
> I reverted it back. I'll swap it back to the way it was and disable the lint if you want me to.

I'm fine with either version. I think both are correct, it's possible one is more strict than the other (and hence better) but it's unclear which one (since the lint has reportedly not been very reliable).

Reply:
> Ah yeah, had it in my mailbox, just accepted. Thanks.

I *think* that gives you full control. I'll dig through the settings again to make sure thats the case.

Reply:
[insert spiderman quote]

Well I'm happy just being able to resolve my own threads :p